### PR TITLE
Add user agent

### PR DESCRIPTION
**Pull request by AllanFly120**
I added 2 packages: util-user-agent-node and util-user-agent-browser, which will be imported accordingly when generating node and browser service clients. The 2 packages will provide default user-agent value by a middleware in the service client.

The last commit is regenerating service clients. When reviewing you can ignore the changes from the laster commit. But the genrated service clients give you the idea of how user-agent is preserved.

### CODE REVIEW

File: packages/service-types-generator/src/ServiceCustomizations/customizationsFromModel/userAgentCustomization.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        imports.push(IMPORTS['util-user-agent-browser']);
+    }
+    const defaultValuePackageName = runtime === 'node' ? '@aws-sdk/util-user-agent-node' : '@aws-sdk/util-user-agent-browser';
+    return {
+        client: [{
+            type: 'Middleware',
+            step: 'build',
+            priority: 0,
+            imports,
+            expression: `${packageNameToVariable('@aws-sdk/middleware-header-default')}.headerDefault({'${runtime === 'node' ? 'User-Agent' : 'X-Amz-User-Agent'}': ${packageNameToVariable(defaultValuePackageName)}.defaultUserAgent(ServiceMetadata.serviceId || ServiceMetadata.endpointPrefix, clientVersion)})`,
```
Comment (seebees) [inline packages/service-types-generator/src/ServiceCustomizations/customizationsFromModel/userAgentCustomization.ts]: This seems overly complicated and long for a single line.

File: packages/client-xray-node/types/GetSamplingTargetsOutput.ts [hunk 2]
```diff
@@ -7,17 +7,17 @@ import * as __aws_sdk_types from '@aws-sdk/types';
...
 export interface GetSamplingTargetsOutput {
     /**
-     * _SamplingTargetDocumentList shape
+     * <p>Updated rules that the service should use to sample requests.</p>
      */
     SamplingTargetDocuments?: Array<_UnmarshalledSamplingTargetDocument>;
 
     /**
-     * _Timestamp shape
+     * <p>The last time a user changed the sampling rule configuration. If the sampling rule configuration changed since the service last retrieved it, the service should call <a>GetSamplingRules</a> to get the latest version.</p>
```
[the remark this answers is not in the document]
Comment (AllanFly120) [reply packages/client-xray-node/types/GetSamplingTargetsOutput.ts]: These docs are used to generated in TS Doc, which will be much easier to read. It also doesn't affect intellisense, so I guess this is fine.

File: packages/client-xray-node/types/PutTelemetryRecordsInput.ts [hunk 3]
```diff
@@ -7,22 +7,22 @@ import * as __aws_sdk_types from '@aws-sdk/types';
  */
 export interface PutTelemetryRecordsInput {
     /**
-     * _TelemetryRecordList shape
+     * <p/>
```
[the remark this answers is not in the document]
Comment (AllanFly120) [reply packages/client-xray-node/types/PutTelemetryRecordsInput.ts]: These are generated from service model, will sync with service team on this issue.

File: packages/service-types-generator/src/Components/Client/Client.ts [hunk 4]
```diff
@@ -62,6 +62,7 @@ export class Client {
 ${configurationImports.toString()}
 import {InputTypesUnion} from './types/InputTypesUnion';
 import {OutputTypesUnion} from './types/OutputTypesUnion';
+import {clientVersion, ServiceMetadata} from './model/ServiceMetadata';
```
[the remark this answers is not in the document]
Comment (AllanFly120) [reply packages/service-types-generator/src/Components/Client/Client.ts]: These will be used in the client for adding user-agent header. If you see the generated service clients you will see these imports are always referred.

**Comment by srchase** [comment]
The additional changes on this look good!

**Pull request merged**